switchsdk/src/androidTest/java/net/gini/switchsdk/uitests/ReviewPictureScreenTest.java

```diff
@@ -0,0 +1,162 @@
+import android.support.test.filters.LargeTest;
+import android.support.test.rule.ActivityTestRule;
+import android.support.test.rule.GrantPermissionRule;
+import android.support.test.runner.AndroidJUnit4;
+import android.view.View;
+
+import net.gini.switchsdk.OnboardingManager;
+import net.gini.switchsdk.R;
+import net.gini.switchsdk.SwitchSdk;
+
+import org.hamcrest.core.IsInstanceOf;
+import org.junit.After;
+import org.junit.Before;
+import org.junit.Rule;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+
+@RunWith(AndroidJUnit4.class)
+@LargeTest
+public class ReviewPictureScreenTest {
+    private static final int TIME_TO_WAIT_BETWEEN_STEPS = 1000;
+    @Rule
+    public ActivityTestRule<HostActivity> mActivityTestRule = new ActivityTestRule
+            <HostActivity>(HostActivity.class) {
+        @Override
+        protected Intent getActivityIntent() {
+            Context context = InstrumentationRegistry.getInstrumentation().getTargetContext();
+            SwitchSdk switchSdk = SwitchSdk.init(context, "", "", "");
+            Intent intent = new Intent(context, HostActivity.class);
+
+            intent.putExtras(switchSdk.getSwitchSdkIntent());
+            return intent;
+        }
+    };
+    @Rule
+    public GrantPermissionRule mRuntimePermissionRule = GrantPermissionRule.grant(
+            Manifest.permission.CAMERA);
+
+    @Test
+    public void menu_cancelShouldDisplayDialog() {
+
+        openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+        ViewInteraction appCompatTextView = onView(
+                allOf(withId(R.id.title), withText(R.string.menu_entry_cancel), isDisplayed()));
+        appCompatTextView.perform(click());
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+
+        onView(withText(R.string.exit_dialog_text))
+                .inRoot(isDialog())
+                .check(matches(isDisplayed()));
+
+        onView(withId(android.R.id.button1))
+                .inRoot(isDialog())
+                .check(matches(withText(R.string.confirm)))
+                .check(matches(isDisplayed()));
+
+        onView(withId(android.R.id.button2))
+                .inRoot(isDialog())
+                .check(matches(withText(R.string.cancel)))
+                .check(matches(isDisplayed()));
+
+    }
+
+    @Test
+    public void menu_helpShouldDisplayOnboarding() {
+
+        openActionBarOverflowOrOptionsMenu(getInstrumentation().getTargetContext());
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+        ViewInteraction appCompatTextView = onView(
+                allOf(withId(R.id.title), withText(R.string.menu_entry_help), isDisplayed()));
+        appCompatTextView.perform(click());
+
+        SystemClock.sleep(TIME_TO_WAIT_BETWEEN_STEPS);
+
+        onView(withId(R.id.onBoardingViewPager)).check(matches(isDisplayed()));
+
+    }
+
+    @Test
+    public void menu_shouldContainAllItems() {
```

Comment:
Since those are the review screen tests, I wouldn't put so much emphasis on the overflow menu. There are 3 tests about that menu and only one about the actual review screen.